experimental/table/server/common/src/main/java/alluxio/table/common/udb/UdbConfiguration.java

```diff
@@ -12,24 +12,69 @@
 package alluxio.table.common.udb;
 
 import alluxio.table.common.BaseConfiguration;
+import alluxio.table.common.ConfigurationUtils;
 
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.util.Collections;
+import java.util.HashMap;
 import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 /**
  * This represents a configuration of the UDB.
  */
 public class UdbConfiguration extends BaseConfiguration<UdbProperty> {
   private static final Logger LOG = LoggerFactory.getLogger(UdbConfiguration.class);
 
+  protected final Map<String, Map<String, String>> mMountOptions;
+
+  protected UdbConfiguration() {
+    super();
+    mMountOptions = new HashMap<>();
+  }
+
   /**
    * Creates an instance.
    *
    * @param values the map of values
    */
   public UdbConfiguration(Map<String, String> values) {
     super(values);
+    mMountOptions = new HashMap<>(values.size());
+    String mountPrefix = ConfigurationUtils.MOUNT_PREFIX;
+    Pattern datePatt = Pattern.compile("(\\(.*\\))\\.(.+?)");
+    for (Map.Entry<String, String> entry : values.entrySet()) {
+      if (entry.getKey().startsWith(mountPrefix)) {
+        String key = entry.getKey().substring(mountPrefix.length());
+        Matcher m = datePatt.matcher(key);
+        if (m.matches()) {
+          String resource = m.group(1);
+          // remove the bracket around the scheme://authority
+          resource = resource.substring(1, resource.length() - 1);
```

Comment:
This feels a little sketchy. Regex might be a better option - but also could be overkill

Reply:
i think Regex might be overkill in this case, we already extracted the substring using regex so it must begin with a bracket and end with a bracket.

Reply:
What if someone made a typo

`udb.mountoption.([hdfs://host/path).mountoption.readonly`?

Reply:
i think in that case, the mountoption will not apply to the mount operation. I don't see how we can guard against user error in that case though.. we can only assume whatever in the bracket is the uri.